src/ncp/ncp_spinel.cpp

```diff
@@ -110,6 +110,25 @@ void NcpSpinel::DatasetSetActiveTlvs(const otOperationalDatasetTlvs &aActiveOpDa
     }
 }
 
+void NcpSpinel::DatasetMgmtSetPending(const otOperationalDatasetTlvs &aPendingOpDatasetTlvs, AsyncTaskPtr aAsyncTask)
+{
+    otError      error        = OT_ERROR_NONE;
+    EncodingFunc encodingFunc = [this, &aPendingOpDatasetTlvs] {
```

Comment:
+1 on same question?

Reply:
The call stack is: DBus API -> ThreadHost API ->  NcpSpinel API -> encodingFunc(pendingDataset)

And the DBus API will only exit after calling the whole stack, including the encoding function. So the variable is still valid. 

I agree it's a little confusing here. Here an encoding func is used in `SetProperty` NOT for async purposes but just for reusing the code of setting properties (get a tid, check if there is an available tid, encode a header, then encode the body, send the frame, etc). Another approach would be using templates: we templatize `SetProperty` and use template specialization for encoding the body. But I think that leads to an increasing of code size. Thoughts?

Reply:
IMHO we may consider passing in `aPendingOpDatasetTlvs` by value or store it into a `shared_ptr` and pass it in by value so that we can feel safer when reviewing/updating the code.

Reply:
SG! Updated.

Reply:
`shared_ptr` works nicely. 👍